sdks/python/apache_beam/dataflow_test.py

```diff
@@ -109,11 +109,12 @@ def test_par_do_with_side_input_as_keyword_arg(self):
 
   @attr('ValidatesRunner')
   def test_par_do_with_do_fn_object(self):
-    class SomeDoFn(DoFn):
+    class SomeDoFn(NewDoFn):
       """A custom DoFn for a FlatMap transform."""
 
-      def process(self, context, prefix, suffix):
-        return ['%s-%s-%s' % (prefix, context.element, suffix)]
+      def process(self, prefix, element=NewDoFn.ElementParam,
```

Comment:
1. No need. The "defaults" should be (element, side_input, side_input, ...). Actually, I wouldn't be opposed to requiring all side inputs to come last to be in the same order as the ParDo call (the interleaving may be odd). This can be relaxed, but not imposed, in the future based on more experience. 

2. No, we should not differentiate between these cases.